includes/class-donations.php

```diff
@@ -65,17 +65,19 @@ class Donations {
 	 */
 	public static function init() {
 		self::$donation_product_name = __( 'Donate', 'newspack' );
-		if ( ! is_admin() ) {
-			add_action( 'wp_loaded', [ __CLASS__, 'process_donation_form' ], 99 );
-			add_action( 'woocommerce_checkout_update_order_meta', [ __CLASS__, 'woocommerce_checkout_update_order_meta' ] );
-			add_filter( 'woocommerce_billing_fields', [ __CLASS__, 'woocommerce_billing_fields' ] );
-			add_filter( 'pre_option_woocommerce_enable_guest_checkout', [ __CLASS__, 'disable_guest_checkout' ] );
-			add_action( 'woocommerce_check_cart_items', [ __CLASS__, 'handle_cart' ] );
-			add_filter( 'amp_skip_post', [ __CLASS__, 'should_skip_amp' ], 10, 2 );
-			add_filter( 'newspack_blocks_donate_billing_fields_keys', [ __CLASS__, 'get_billing_fields' ] );
-			add_action( 'woocommerce_checkout_create_order_line_item', [ __CLASS__, 'checkout_create_order_line_item' ], 10, 4 );
-			add_action( 'woocommerce_coupons_enabled', [ __CLASS__, 'disable_coupons' ] );
-		}
+
```

Comment:
I'm not sure if this is related to removing this block, but the warning is coming from this file. On this branch I see a lot of this warning in debug.log:

```
wc_get_product was called incorrectly. wc_get_product should not be called before the woocommerce_init, woocommerce_after_register_taxonomy and woocommerce_after_register_post_type actions have finished. Backtrace: require_once('wp-load.php'), require_once('wp-config.php'), require_once('wp-settings.php'), do_action('init'), WP_Hook->do_action, WP_Hook->apply_filters, Automattic\WooCommerce\Admin\Features\Features::load_features, Automattic\WooCommerce\Internal\Admin\Coupons->__construct, Automattic\WooCommerce\Internal\Admin\Notes\CouponPageMoved->init, wc_coupons_enabled, apply_filters('woocommerce_coupons_enabled'), WP_Hook->apply_filters, Newspack\Donations::disable_coupons, Newspack\Donations::is_donation_cart, Newspack\Donations::get_donation_product_child_products_ids, Newspack\Donations::get_parent_donation_product, wc_get_product, wc_doing_it_wrong. This message was added in version 3.9.
```

Reply:
From your stack, looks like it's coming from here:

https://github.com/Automattic/newspack-plugin/blob/1dcfd82d04755c5f781bc79195dfcd095f526ac9/includes/class-donations.php#L1047-L1063

I'll take a closer look tomorrow

Reply:
Found it! Here's a PR with the fix: https://github.com/Automattic/newspack-plugin/pull/2822

Pushing the fix separately as it's unrelated to RAS-ACC and should be out sooner.

Reply:
Aha, I see! Thanks for the quick turnaround. However, I don't think we need to push #2822 separately to `master` as we're only removing the [`if ( ! is_admin() )`](https://github.com/Automattic/newspack-plugin/blob/master/includes/class-donations.php#L68) condition in this PR.

Reply:
Of course! Updated it here with b0c4828c2869456a01d0a9f37468573a2c778ceb